### PR TITLE
Made fiscalization checker parse instead of validating.

### PR DESCRIPTION
**Pull request by KaliCZ**
Created a Dto folder and Model folder and separated logic into clear 
- reading
- parsing
- Control of the flow is now completely in Program.cs

### CODE REVIEW

File: src/Extensions.cs [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
+        internal static string ToSignatureString(this Amount amount)
+        {
+            return ((int)(amount.Value * 100)).ToString();
+        }
+
+        internal static string ToSignatureString(this TaxSummary taxSummary)
+        {
+            var parts = taxSummary.Data.OrderByDescending(d => d.Key).Select(d => $"{d.Key.ToSignatureString()}:{d.Value.ToSignatureString()}");
+            return String.Join("|", parts);
+        }
```
Comment (marektresnak) [inline src/Extensions.cs:34]: I would move this to the class directly. Or is there a reason to have them as extensions?

File: src/Model/ArchiveMetadata.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            TerminalIdentification = terminalIdentification;
+            PreviousRecordSignature = previousRecordSignature;
+            Created = created;
+            Version = version;
+        }
+
+        public string TerminalIdentification { get; }
+
+        public IOption<string> PreviousRecordSignature { get; }
```
Comment (marektresnak) [inline src/Model/ArchiveMetadata.cs:20]: Shouldn't this be of `Signature` type?

File: src/ArchiveReader.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        private static ITry<ArchiveEntry, IEnumerable<string>> ReadEntry(IReadOnlyList<ArchiveEntry> archiveEntries, string namePrefix)
+        {
+            return ReadOptionalEntry(archiveEntries, namePrefix).ToTry(_ => $"No unique file found {namePrefix}*.".ToEnumerable());
+        }
+
+        private static IOption<ArchiveEntry> ReadOptionalEntry(IReadOnlyList<ArchiveEntry> archiveEntries, string namePrefix, bool isOptional = false)
+        {
+            return archiveEntries.SingleOption(e => e.Name.StartsWith(namePrefix));
+        }
```
Comment (marektresnak) [inline src/ArchiveReader.cs:34]: Nit: this is more like `GetEntry`?

File: src/Program.cs [hunk 4]
```diff
@@ -4,59 +4,58 @@
...
-            var archive = args.SingleOption().Match(
-                a => Archive.Load(a),
-                _ => Try.Error<Archive, string>("Invalid arguments")
-            );
-            var result = archive.FlatMap(a => IsArchiveValid(a)).Match(
-                r => r.Match(
+            var path = args.SingleOption().ToTry(_ => "Invalid arguments".ToEnumerable());
+            var archiveEntries = path.FlatMap(p => ZipArchiveReader.ReadArchive(p));
+            var archiveFileContent = archiveEntries.FlatMap(e => ArchiveReader.ReadArchive(e));
+            var archive = archiveFileContent.FlatMap(a => Archive.Create(a));
```
[the remark this answers is not in the document]
Comment (KaliCZ) [reply src/Program.cs:18]: We're talking about using this in Mews, so I'd say there we will create the archive Entries from the stringContent in memory, so imo useful.

File: src/ZipArchiveReader.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        private static byte[] ReadFully(Stream stream, bool seekToBeginning = false)
+        {
+            if (seekToBeginning)
+            {
+                stream.Seek(0, SeekOrigin.Begin);
+            }
+
+            var buffer = new byte[32768];
```
[the remark this answers is not in the document]
Comment (KaliCZ) [reply src/ZipArchiveReader.cs:51]: Sure, then in a separate PR that actually changes logic.

**Review by marektresnak — APPROVED** [review]

**Pull request merged**